tokio/src/sync/notify.rs

```diff
@@ -1138,7 +1205,9 @@ impl Drop for Notified<'_> {
             // the notification was triggered via `notify_one`, it must be sent
             // to the next waiter.
             if notification == Some(Notification::One) {
-                if let Some(waker) = notify_locked(&mut waiters, &notify.state, notify_state) {
+                if let Some(waker) =
+                    notify_locked(&mut waiters, &notify.state, notify_state, strategy.unwrap())
```

Comment:
I think we can drop the `Option` around the strategy. Then we don't need an `unwrap` here. We can ignore the strategy when we are waking everything.

Reply:
Would you prefer to just pass or return the `NOTIFICATION_NOTIFY_ONE_STRATEGY_NONE`?  If so we can also avoid this two indirection using the const and the enum and having somthing like

```
 enum NotifyOneStrategy {
    None = 0
    Fifo = 1,
    Lifo = 2,
}
```

And use the `NotifyOneStrategy` as a parameter for the [store](https://github.com/tokio-rs/tokio/pull/6520/files#diff-e6a0a252981328ec25e728ff0cf7c30125c33e85a73667d31c971e8efecab68eR292) and as a return value for the [load](https://github.com/tokio-rs/tokio/pull/6520/files#diff-e6a0a252981328ec25e728ff0cf7c30125c33e85a73667d31c971e8efecab68eR304).

Reply:
I commented above, but it looks like the strategy is an argument to `Notification`. Something like:

```rust
enum Notification {
    One(Strategy),
    All,
}
```

You can implement the conversion of the enum to size using a match statement, the compiler will optimize it anyway.

Reply:
Yeps, changing a bit this one.